openlibrary/plugins/upstream/account.py

```diff
@@ -692,8 +692,10 @@ def GET(self):
 
     @require_login
     def POST(self):
+        i = web.input(public_readlog="", safe_mode="")
         user = accounts.get_current_user()
-        user.save_preferences(web.input())
+        user.save_preferences(i)
+        web.setcookie('sfw', i.safe_mode, expires="" if i.safe_mode == 'yes' else -1)
```

Comment:
Just to guard against future changes to HTML files.
```suggestion
        web.setcookie('sfw', i.safe_mode, expires="" if i.safe_mode.lower() == 'yes' else -1)
```